### ISSUE TITLE
font sizes too small

### ISSUE THREAD

**Issue by ppiz** [LMMS/lmms#2209]
For the modern big screens the font sizes are just to small.
It needs to be adjustable at least some sizes should be available.

**Comment by michaelgregorius** [comment]
I have committed a first fix with pull request #2217. This fix solves the problem of too small fonts for all file browsers ("My Projects", "My Samples", "My Presets", etc.).

I think most of the problems with too small fonts on high DPI displays are caused by calls to `pointSize` and `pointSizeF`. The intention of these functions seems to be to create a font that will always have the same height in pixels regardless of the actual DPI value of the display. However, it is not a good idea to have fonts that are 8 pixels in height on 4K displays. So in my opinion we should try to get rid of these calls and try to make all widgets dynamically sizeable.

**Comment by Wallacoloo** [comment]
@michaelgregorius after #2217 and #2222, are there still any places in which we need to fix the fonts? If so, we should make a list of them and turn this into a WIP. Otherwise, this issue should be resolved.

**Comment by michaelgregorius** [comment]
@Wallacoloo Yes, there are still several places where the fixed font sizes are used. For example in the song editor's tracks, also in lots of plugins and other places where custom widgets are used. I'd say there is still tons of work to do.

Here is a list of places where `pointSize` and `pointSizeF` are still called:
```
$ grep -l -r pointSize | sort | uniq
include/gui_templates.h
plugins/audio_file_processor/audio_file_processor.cpp
plugins/carlabase/carla.cpp
plugins/DualFilter/DualFilterControlDialog.cpp
plugins/ladspa_browser/ladspa_browser.cpp
plugins/monstro/Monstro.cpp
plugins/patman/patman.cpp
plugins/stk/mallets/mallets.cpp
plugins/vestige/vestige.cpp
plugins/VstEffect/VstEffectControlDialog.cpp
plugins/zynaddsubfx/ZynAddSubFx.cpp
src/core/audio/AudioAlsa.cpp
src/core/audio/AudioJack.cpp
src/core/audio/AudioOss.cpp
src/core/audio/AudioPortAudio.cpp
src/core/audio/AudioPulseAudio.cpp
src/core/audio/AudioSdl.cpp
src/core/midi/MidiAlsaRaw.cpp
src/core/midi/MidiAlsaSeq.cpp
src/core/midi/MidiOss.cpp
src/core/Track.cpp
src/gui/AutomationPatternView.cpp
src/gui/editors/AutomationEditor.cpp
src/gui/editors/PianoRoll.cpp
src/gui/PianoView.cpp
src/gui/SetupDialog.cpp
src/gui/widgets/ComboBox.cpp
src/gui/widgets/EffectView.cpp
src/gui/widgets/EnvelopeAndLfoView.cpp
src/gui/widgets/FxLine.cpp
src/gui/widgets/GroupBox.cpp
src/gui/widgets/InstrumentFunctionViews.cpp
src/gui/widgets/InstrumentMidiIOView.cpp
src/gui/widgets/InstrumentSoundShapingView.cpp
src/gui/widgets/Knob.cpp
src/gui/widgets/LcdWidget.cpp
src/gui/widgets/LedCheckbox.cpp
src/gui/widgets/MeterDialog.cpp
src/gui/widgets/MidiPortMenu.cpp
src/gui/widgets/ProjectNotes.cpp
src/gui/widgets/SideBarWidget.cpp
src/gui/widgets/TabBar.cpp
src/gui/widgets/TabWidget.cpp
src/gui/widgets/TextFloat.cpp
src/gui/widgets/VisualizationWidget.cpp
src/tracks/BBTrack.cpp
src/tracks/InstrumentTrack.cpp
src/tracks/Pattern.cpp
src/tracks/SampleTrack.cpp
```
Do you want to open a new issue for some of the remaining changes or reuse this one?

**Comment by Wallacoloo** [comment]
:+1: thanks for that list. Let's continue to use this issue & we can just update the list of places where fixed fonts are being used after each PR until it's completely fixed :)

**Comment by tresf** [comment]
:+1: :+1:

**Comment by LYF610400210** [comment]
@Wallacoloo The number in the piano roll and the multi-track window is too small.
![font](https://cloud.githubusercontent.com/assets/6359921/9511472/c5e4d67e-4ca9-11e5-9c4d-f92bec6543ca.PNG)

**Comment by tresf** [comment]
Duplicate of #2510.  Let's track it there.

Will quote/copy&paste @michaelgregorius's findings into the other thread so we don't loose it.
